### PR TITLE
fix gas benchmarks

### LINKED ISSUE #3733: Fix the gas benchmarks (exit format branch)

### PR DESCRIPTION
**Pull request by geoknee**
fixes #3733 

~warning: these benchmarks do not assert on correct behaviour. While we wait for other tests to do that, these numbers should be taken with a grain of salt. They seem a little too good to be true to me.~

### CODE REVIEW

File: packages/nitro-protocol/gas-benchmarks/benchmark.test.ts [hunk 1]
```diff
@@ -7,8 +7,7 @@ import {
   Y,
   X,
   LforX,
-  LforG,
-  G,
```
Comment (NiloCK) [inline packages/nitro-protocol/gas-benchmarks/benchmark.test.ts:11]: :+1:

File: packages/nitro-protocol/contracts/NitroAdjudicator.sol [hunk 2]
```diff
@@ -129,4 +129,15 @@ contract NitroAdjudicator is ForceMove, MultiAssetHolder {
...
+    /**
+     * @notice Executes an exit by paying out assets and calling external contracts
+     * @dev Executes an exit by paying out assets and calling external contracts
+     * @param exit The exit to be paid out.
+     */
+    function executeExit(Outcome.SingleAssetExit[] memory exit) internal {
+        for (uint256 assetIndex = 0; assetIndex < exit.length; assetIndex++) {
+            executeSingleAssetExit(exit[assetIndex]);
+        }
+    }
```
Comment (NiloCK) [inline packages/nitro-protocol/contracts/NitroAdjudicator.sol:142]: two nits:

For consistency:
```suggestion
    function _executeExit(Outcome.SingleAssetExit[] memory exit) internal {
        for (uint256 assetIndex = 0; assetIndex < exit.length; assetIndex++) {
            _executeSingleAssetExit(exit[assetIndex]);
        }
    }
```
And a question: is this function better grouped with executeSingleAssetExit in MAH? Maybe not - the current location is better at minimizing API size along the chain.

File: packages/nitro-protocol/gas-benchmarks/gas.ts [hunk 3]
```diff
@@ -87,29 +87,26 @@ export const gasRequiredTo: GasRequiredTo = {
...
+      // initially                   ⬛ ->  L  ->  J  ->  X  -> 👩
+      // challenge L,J,X + timeout   ⬛ -> (L) -> (J) -> (X) -> 👩
+      // claimL                      ⬛ ---------------> (X) -> 👩
+      // transferAllAssetsX          ⬛ ----------------------> 👩
+      challengeL: 94968,
+      challengeJ: 103014,
+      challengeX: 94673,
+      claimL: 107224,
+      transferAllAssetsX: 55606,
+      total: 455485,
```
Comment (NiloCK) [inline packages/nitro-protocol/gas-benchmarks/gas.ts:109]: :eyes:

**Review by NiloCK — APPROVED** [review]
Hooray! (Grain of salt notwithstanding)

File: packages/nitro-protocol/gas-benchmarks/fixtures.ts [hunk 4]
```diff
@@ -164,34 +178,66 @@ export const Y = new TestChannel(
...
+  [Alice, Bob],
+  [
+    {
+      destination: J.channelId,
+      amount: '0xa',
+      metadata: encodeGuaranteeData([
+        X.channelId,
+        convertAddressToBytes32(Alice.address),
+        convertAddressToBytes32(Ingrid.address),
+      ]),
```
Comment (geoknee) [inline packages/nitro-protocol/gas-benchmarks/fixtures.ts:236]: Question: this looks incorrect on master?

**Pull request merged**